### ISSUE TITLE
I think mipmaps are not being used

### ISSUE THREAD

**Issue by dortamiguel** [David-DiGioia/vulkan-renderer#1]
Hi, I was looking at your implementation for mipmaps and I noticed this line

https://github.com/David-DiGioia/monet/blob/main/src/render_to_texture.cpp#L568

I think `sampler.maxLod = 1.0f;` should be instead `sampler.maxLod = mipLevels;`

From https://vulkan-tutorial.com/Generating_Mipmaps it says:
```
To allow the full range of mip levels to be used, we set minLod to 0.0f, and maxLod to the number of mip levels. We have no reason to change the lod value , so we set mipLodBias to 0.0f.
```

**Comment by David-DiGioia** [comment]
This is the sampler for the shadow depth buffer so it does not use mipmaps actually, since we always want to sample from the full resolution depth map. Although why `1.0f` is written there instead of `0.0f` had me a little confused, but I found that I actually was copying Sascha Willems example for that piece of code:

https://github.com/SaschaWillems/Vulkan/blob/ec7c76d81220e91e5dbff494ee818a12857e5cb5/examples/shadowmapping/shadowmapping.cpp#L247

So why did Sascha Willems write `1.0f` there? I don't know, but I found [a stack overflow answer](https://stackoverflow.com/questions/51659923/can-a-vulkan-samplers-maxlod-be-too-high) that says

> The actual mipmap level accessed is the computed lod (clamped to maxLod), further clamped to the number of mipmaps levels actually available in the texture.

So the mipmap level will get clamped to 0 regardless of maxLod's value since the texture itself only has a level 0 miplevel.

(I didn't know anyone looked at this project! It's not very well organized and it may not be the best project to learn from. I recently started a new game engine called [Pumpkin](https://github.com/David-DiGioia/pumpkin) which I'm architecting better, if you're interested)
